3rdparty/openvx/src/openvx_hal.cpp

```diff
@@ -0,0 +1,156 @@
+#include "openvx_hal.hpp"
+
+
+//OpenVX includes
+#include <VX/vx.h>
+#include <VX/vxu.h>
+#include <VX/vx_lib_debug.h>
+#include <VX/vx_helper.h>
+#include <VX/vx_lib_extras.h>
+
+vxContext * vxContext::getContext()
+{
+    // not thread safe
+    static vxContext instance;
+    return &instance;
+}
+
+void openvx_hal_init()
+{
+    //do nothing
+}
+
+#define OVX_BINARY_OP(hal_name, hal_type, ovx_type, ovx_func, ...)                                                                                      \
+int ovx_hal_##hal_name(const hal_type* src1, size_t sz1, const hal_type* src2, size_t sz2, hal_type* dst, size_t sz, int w, int h, __VA_ARGS__)         \
+{                                                                                                                                                       \
+    vx_status status;                                                                                                                                   \
+    vx_context context = vxCreateContext(); /* Context is common to all calls. So it could be better to create it in hal_init and release in hal_free*/ \
+    status = vxGetStatus((vx_reference)context);                                                                                                        \
+    if (status != VX_SUCCESS)                                                                                                                           \
+        return CV_HAL_ERROR_UNKNOWN; /* Unable to initialize OpenVX. Something goes wrong.*/                                                            \
```

Comment:
For now we do not have any calls for this function, but use single context instance via the vxContext class.